src/main/java/com/github/luben/zstd/ZstdCompressCtx.java

```diff
@@ -268,6 +276,117 @@ public ZstdCompressCtx setLong(int windowLog) {
         return this;
     }
 
+    /**
+     * Register an external sequence producer
+     * @param producer the user-defined {@link SequenceProducer} to register.
+     */
+    public ZstdCompressCtx registerSequenceProducer(SequenceProducer producer) {
+        ensureOpen();
+        acquireSharedLock();
+        try {
+            if (this.seqprod != null) {
+                this.seqprod.freeState(seqprod_state);
+                this.seqprod = null;
+            }
+
+            if (producer == null) {
+                Zstd.registerSequenceProducer(nativePtr, 0, 0);
+            } else {
+                seqprod_state = producer.createState();
+                Zstd.registerSequenceProducer(nativePtr, seqprod_state, producer.getFunctionPointer());
+                this.seqprod = producer;
+            }
+        } catch (Exception e) {
```

Comment:
Where this exception may come? Is it possible to narrow the exception type? Ideally using `ZstdException`.

Reply:
`registerSequenceProducer()` doesn't return any errors, this block is mainly to protect against UAF if the user-provided `producer.createState()` fails somehow (since `seqprod_state` has been freed and the old producer is still registered).